src/components/GenerateCVButton/container.js

```diff
@@ -0,0 +1,18 @@
+import React from 'react';
+
+export default Wrapped =>
+  class extends React.Component {
+    isDisabled = () => {
+      // logic added in next PR
+      return false;
+    };
+
+    handleOnClick = () => {
+      // logic added in next PR
+      console.log('click');
+    };
+
+    render() {
+      return <Wrapped disabled={this.isDisabled()} handleClick={() => this.handleOnClick()} />;
```

Comment:
Could you not just pass down `handleClick={this.handleOnClick}` then call the function once in the component it should be invoked in? Everyone has their own pattern but I personally find it cleaner that way.

Reply:
Garry I actually refactored this back to my original solution of having the click logic inside the Button component rather than the page. 

My reasoning was that in the next PR the Page component started to look very bloated with logic belonging to its child components. If I take the logic out for handling the click then I also have to take out the logic for disabling the button and do the same thing for other child components of the Page. The result didn't look like a good separation of concerns.